internal/generator/gen.go

```diff
@@ -118,18 +142,24 @@ type output struct {
 	template     string // template file name
 	data         data
 	withPreamble bool
+	options      *options
 }
 
-func newOutput(name, template string, data data, withPreamble bool) output {
+func newOutput(name, template string, templateData data, withPreamble bool, opts *options) output {
```

Comment:
```suggestion
func newOutput(name, template string, templateData data, opts *options) output {
```

Maybe withPreamble can be part of the opts? the struct attribute on `output` can stay, if we want.

Reply:
I wanted to do that first, but decided against it since the preamble flag is very much an internal thing depending on the specific output you are generating, and not part of the global options. You can see this when you look at `newOutputs()`, which uses ` !opts.outputToStdout` for the `withPreamble` flag most of the time, but at the very end unconditionally sets it to `true` when generating `pkg.go`.